docs/gen-ai/gen-ai-metrics.md

```diff
@@ -171,6 +175,106 @@ Instrumentations SHOULD document the list of errors they report.
+
+### Metric: `gen_ai.server.latency.request_duration`
+
+This metric is [recommended][MetricRecommended] to report the model server
+latency in terms of time spent per request.
+
+For example, if a GenAI model server reports latency information, it SHOULD be
+used.
+
+If instrumentation cannot obtain this information at a request level and break
+it down into the buckets mentioned below, then it MUST NOT report this metric.
+
+This metric SHOULD be specified with [ExplicitBucketBoundaries] of
+[1.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0, 40.0, 50.0, 60.0].
+
+<!-- semconv metric.gen_ai.server.latency.request_duration(metric_table) -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `gen_ai.server.latency.request_duration` | Histogram | `s` | Time (end to end latency) to complete a request | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+<!-- END AUTOGENERATED TEXT -->
+<!-- endsemconv -->
+
+### Metric: `gen_ai.server.latency.time_per_output_token`
+
+This metric is [recommended][MetricRecommended] to report the model server
+latency in terms of time per token generated for any model servers which
+support serving LLMs.
+
+For example, if a model server which serves LLMs reports latency information,
+it SHOULD be used.
+
+If instrumentation cannot obtain this information at a request level and break
+it down into the buckets mentioned below, then it MUST NOT report this metric.
+
+This metric SHOULD be specified with [ExplicitBucketBoundaries] of
+[0.01, 0.025, 0.05, 0.075, 0.1, 0.15, 0.2, 0.3, 0.4, 0.5, 0.75, 1.0, 2.5].
+
+<!-- semconv metric.gen_ai.server.latency.time_per_output_token(metric_table) -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `gen_ai.server.latency.time_per_output_token` | Histogram | `s` | Time per output token generated for successful responses | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+<!-- END AUTOGENERATED TEXT -->
+<!-- endsemconv -->
+
+### Metric: `gen_ai.server.latency.time_to_first_token`
+
+This metric is [recommended][MetricRecommended] to report the model server
+latency in terms of time spent to generate the first token of the response for
```

Comment:
should it be reported if token is never generated? I.e. should the metric have `error.type` attribute or it only captures successful generations?
similar question for `time_per_output_token`

Reply:
Both of these are cumulative and report only on successful token generations.